src/region.js

```diff
@@ -33,155 +33,108 @@ var Region = MarionetteObject.extend({
     MarionetteObject.call(this, options);
   },
 
-  // Displays a backbone view instance inside of the region.
-  // Handles calling the `render` method for you. Reads content
-  // directly from the `el` attribute. Also calls an optional
-  // `onShow` and `onDestroy` method on your view, just after showing
-  // or just before destroying the view, respectively.
-  // The `preventDestroy` option can be used to prevent a view from
-  // the old view being destroyed on show.
-  // The `forceShow` option can be used to force a view to be
-  // re-rendered if it's already shown in the region.
+  // Displays a backbone view instance inside of the region. Handles calling the `render`
+  // method for you. Reads content directly from the `el` attribute. The `preventDestroy`
+  // option can be used to prevent a view from the old view being destroyed on show.
+  // The `forceShow` option can be used to force a view to be re-rendered if it's already
+  // shown in the region.
   show: function(view, options) {
     if (!this._ensureElement()) {
       return;
     }
+    this._ensureView(view);
 
-    this._ensureViewIsIntact(view);
-    MonitorDOMRefresh(view);
-
-    var showOptions     = options || {};
-    var isDifferentView = view !== this.currentView;
-    var forceShow       = !!showOptions.forceShow;
-    var replaceElement  = !!showOptions.replaceElement;
-
-    // We are only changing the view if there is a current view to change to begin with
-    var changingView = this.currentView;
-    var isChangingView = !!changingView;
+    MonitorViewEvents(view);
```

Comment:
I think maybe this should occur either right before or right after the `before:show` gets triggered.  Particularly if the show gets prevented.